packages/components/popover/src/popover.js

```diff
@@ -0,0 +1,185 @@
+import { TSElement, unsafeCSS, html, customElementDefineHelper } from '@tradeshift/elements';
+import '@tradeshift/elements.icon';
+import css from './popover.css';
+
+/**
+ * Position on the X axis.
+ */
+export const PositionX = {
+	LEFT: 'left',
+	RIGHT: 'right'
+};
+
+/**
+ * Position on the Y axis.
+ */
+export const PositionY = {
+	TOP: 'top',
+	BOTTOM: 'bottom'
+};
+
+export class TSPopover extends TSElement {
+	static get styles() {
+		return [TSElement.styles, unsafeCSS(css)];
+	}
+
+	static get properties() {
+		return {
+			/** Is the popover visible or not */
+			opened: { type: Boolean, reflect: true },
+			/** Position on X axis relative to the anchor. Could be 'left', 'right' */
+			positionX: { type: String, attribute: 'position-x', reflect: true },
+			/** Position on Y axis relative to the anchor. Could be 'top', 'bottom' */
+			positionY: { type: String, attribute: 'position-y', reflect: true },
+			/** Text in the title */
+			header: { type: String, reflect: true },
+			/** Anchor element for relative positioning. If you need to position absolutely, leave this empty */
+			anchor: { type: String, reflect: true },
+			/** Left offset when popover is positioned absolutely. Use any valid css syntax for the 'left' property */
+			positionLeft: { type: String, attribute: 'position-left', reflect: true },
+			/** Top offset when popover is positioned absolutely. Use any valid css syntax for the 'top' property */
+			positionTop: { type: String, attribute: 'position-top', reflect: true }
+		};
+	}
+
+	constructor() {
+		super();
+		this.opened = false;
+	}
+
+	onClose() {
+		this.opened = false;
+	}
+
+	stopEventBubbling(e) {
+		e.stopPropagation();
+	}
+
+	attributeChangedCallback(name, oldVal, newVal) {
+		super.attributeChangedCallback(name, oldVal, newVal);
+		switch (name) {
+			case 'opened':
+				if (newVal !== null) {
+					window.customElements.whenDefined('ts-popover').then(() => {
```

Comment:
What's the reason for it? Can this be triggered before the component being defined? 🤔

Reply:
Because we are trying to get the internal DOM node `container`, but it would be unavailable until the element is defined and template is processed. This callback helps with this problem.